test/object-store/sync/flx_sync.cpp

```diff
@@ -4791,6 +4791,159 @@ TEST_CASE("flx: pause and resume bootstrapping at query version 0", "[sync][flx]
     REQUIRE(active_sub_set.state() == sync::SubscriptionSet::State::Complete);
 }
 
+TEST_CASE("flx: collections in mixed - merge lists", "[sync][flx][baas]") {
+    Schema schema{{"TopLevel",
+                   {{"_id", PropertyType::ObjectId, Property::IsPrimary{true}},
+                    {"queryable_str_field", PropertyType::String | PropertyType::Nullable},
+                    {"any", PropertyType::Mixed | PropertyType::Nullable}}}};
+
+    FLXSyncTestHarness::ServerSchema server_schema{schema, {"queryable_str_field"}};
+    FLXSyncTestHarness harness("flx_collections_in_mixed", server_schema);
+    SyncTestFile config(harness.app()->current_user(), harness.schema(), SyncConfig::FLXSyncEnabled{});
+
+    auto set_list_and_insert_element = [](Obj& obj, ColKey col_any, Mixed value) {
+        obj.set_collection(col_any, CollectionType::List);
+        auto list = obj.get_list_ptr<Mixed>(col_any);
+        list->add(value);
+    };
+
+    // Client 1 creates an object and sets property 'any' to an integer value.
+    auto foo_obj_id = ObjectId::gen();
+    harness.load_initial_data([&](SharedRealm realm) {
+        CppContext c(realm);
+        auto obj = Object::create(c, realm, "TopLevel",
```

Comment:
Can you set the any value via the Object::create api rather than doing it separately?